op-node/p2p/sync.go

```diff
@@ -528,34 +544,45 @@ func (s *SyncClient) peerLoop(ctx context.Context, id peer.ID) {
 		// once the peer is available, wait for a sync request.
 		select {
 		case pr := <-s.peerRequests:
+			if !s.activeRangeRequests[pr.rangeReqId] {
+				log.Debug("dropping cancelled p2p sync request", "num", pr.num)
+				delete(s.inFlight, pr.num)
+				continue
+			}
+
 			// We already established the peer is available w.r.t. rate-limiting,
 			// and this is the only loop over this peer, so we can request now.
 			start := time.Now()
+
+			resultCode := byte(0)
 			err := s.doRequest(ctx, id, pr.num)
 			if err != nil {
 				// mark as complete if there's an error: we are not sending any result and can complete immediately.
 				pr.complete.Store(true)
 				log.Warn("failed p2p sync request", "num", pr.num, "err", err)
+
+				if re, ok := err.(requestResultErr); ok {
+					resultCode = re.ResultCode()
+					if re.ResultCode() == 1 { // indicates block not found error
```

Comment:
`resultCode == 1` maps to `ethereum.NotFound` error [here](https://github.com/ethereum-optimism/optimism/blob/5d67d731ee286a8101c8e0228946a5df272b9c1d/op-node/p2p/sync.go#L773)

Reply:
Might be worth extracting a constant that can be shared between the server and client code rather than duplicating.

Reply:
Updated in https://github.com/ethereum-optimism/optimism/pull/10063/commits/585c9224052b16338531bad4b9395ae133ed4b18